### PR TITLE
Camera Recenter, Turn Lane UI Update, Instruction Initialization 

### PR DESCRIPTION
**Pull request by danesfeder**

### CODE REVIEW

File: navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/turnlane/TurnLaneAdapter.java [hunk 1]
```diff
@@ -55,22 +54,24 @@ public void addTurnLanes(IntersectionLanes[] lanes, String maneuverModifier) {
...
   }
 
-  private void setLaneImage(TurnLaneViewHolder holder, IntersectionLanes lane) {
+  private void setLaneImage(TurnLaneViewHolder holder, IntersectionLanes lanes) {
     StringBuilder builder = new StringBuilder();
 
     // Indications
-    if (lane.getIndications() != null) {
-      for (String indication : lane.getIndications()) {
+    if (lanes.getIndications() != null) {
```
Comment (danesfeder) [inline navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/turnlane/TurnLaneAdapter.java]: @ericrwolfe you mind double checking this turn lane logic?

File: navigation/app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/NavigationViewActivity.java [hunk 2]
```diff
@@ -72,26 +178,45 @@ public void onFailure(Call<DirectionsResponse> call, Throwable throwable) {
     Timber.e(throwable.getMessage());
   }
 
+  @SuppressWarnings({"MissingPermission"})
+  private void initLocationEngine() {
+    locationEngine = new LocationSource(this);
```
Comment (cammace) [inline navigation/app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/NavigationViewActivity.java]: Can be replaced with the default navigation one, use `MapboxNavigation.getLocationEngine()`

File: navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/turnlane/TurnLaneAdapter.java (hunk 1 above)
Comment (ericrwolfe) [reply navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/turnlane/TurnLaneAdapter.java]: Can we not show lane guidance if there's only a single lane indication?

Comment (ericrwolfe) [reply navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/turnlane/TurnLaneAdapter.java]: (i.e. only a `straight` arrow or a `left` arrow)

File: navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/turnlane/TurnLaneAdapter.java [hunk 3]
```diff
@@ -55,22 +54,24 @@ public void addTurnLanes(IntersectionLanes[] lanes, String maneuverModifier) {
...
+      for (String indication : lanes.getIndications()) {
         builder.append(indication);
       }
-
-      if (builder.toString().contains(TURN_LANE_INDICATION_STRAIGHT)
-        || (builder.toString().contains(TURN_LANE_INDICATION_NONE) && lane.getValid())) {
+      if (builder.toString().contains(TURN_LANE_INDICATION_STRAIGHT) && lanes.getValid()) {
```
Comment (ericrwolfe) [inline navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/turnlane/TurnLaneAdapter.java]: I'm not sure I understand what this check for `TURN_LANE_INDICATION_STRAIGHT` does.

Comment (danesfeder) [reply navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/turnlane/TurnLaneAdapter.java]: If the lane contains indication `"straight"` and is valid, then we append the modifier, which will allow us to retrieve the asset from the map with multiple arrows and (one dark, one greyed out).  This the issue stemming from not having completely greyed out assets

**Pull request merged**